### ISSUE TITLE
Output a layer while training

### ISSUE THREAD

**Issue by mijosch** [plaidml/plaidml#381]
I dont think it is an issue but a feature.

I am programming a DCGAN atm. and want to output the generators output layer for each batch because i have to generate fake pictures anyway and i dont want to waste the pictures created while learning.

so i want to:
train_on_batch and get the loss as well as the output of the last layer from the generator.

Is that somehow possible?
Thanks for your help

**Comment by tzerrell** [comment]
From your description it sounds like you are running through Keras (let us know if this is wrong). This sort of task sounds like it would be most conveniently handled at the Keras level. Perhaps you could add the output of the last generator layer as one of the model outputs and add the loss of the last generator layer as one of the model metrics. People who work primarily on Keras might be able to help you more with this approach.

If you are directly calling backend functions, you can add a `K.eval(x)` call on the output you want to save and convert the resulting array to an image. You can similarly call `K.eval` on the loss function output you want to save. Using eval like this can be a bit cumbersome and I don't really recommend it unless you can't accomplish what you want with higher level Keras functionality.

Outputting tensor data can degrade performance. Transferring data between the device you are executing on and the host device takes time. In some circumstances, intermediate tensors can be fused or otherwise optimized away, and outputting their values can prevent these optimizations. I don't know if these will have a meaningful or measurable impact in your circumstances, but be aware that this is not guaranteed to be free or trivial in terms of performance.
